### PR TITLE
remove old Jython hacks

### PR DESCRIPTION
**Pull request by mdehoon**
This pull request removes some old Jython hacks.

- [X] I hereby agree to dual licence this and any previous contributions under both
the _Biopython License Agreement_ **AND** the _BSD 3-Clause License_.

- [X] I have read the ``CONTRIBUTING.rst`` file, have run ``pre-commit`` locally,
and understand that AppVeyor and TravisCI will be used to confirm the Biopython unit
tests and style checks pass with these changes.

- [X] I have added my name to the alphabetical contributors listings in the files
``NEWS.rst`` and ``CONTRIB.rst`` as part of this pull request, am listed
already, or do not wish to be listed. (*This acknowledgement is optional.*)

### CODE REVIEW

File: Bio/AlignIO/MafIO.py [hunk 1]
```diff
@@ -35,14 +35,7 @@
 import os
 from itertools import islice
 
-try:
-    from sqlite3 import dbapi2 as _sqlite
-except ImportError:
-    # Not present on Jython, but should be included in Python 2.5
-    # or later (unless compiled from source without its dependencies)
-    # Still want to offer simple parsing/output
-    _sqlite = None
```
Comment (peterjc) [inline Bio/AlignIO/MafIO.py:44]: I've actually had this on a self compiled Python, but that is a rare corner case - and it does complicate the test suite too.

File: Bio/Nexus/Trees.py [hunk 2]
```diff
@@ -594,8 +594,7 @@ def display(self):
         table = [
             ("#", "taxon", "prev", "succ", "brlen", "blen (sum)", "support", "comment")
         ]
-        # Sort this to be consistent across CPython, Jython, etc
-        for i in sorted(self.all_ids()):
```
Comment (peterjc) [inline Bio/Nexus/Trees.py:598]: I think this should still be explicitly sorted - different versions of CPython and PyPy are not consistent.

Comment (mdehoon) [reply Bio/Nexus/Trees.py:598]: Are you sure?

File: Bio/AlignIO/MafIO.py (hunk 1 above)
Comment (mdehoon) [reply Bio/AlignIO/MafIO.py:44]: The thing is that in this module, there is no check if `_sqlite` is `None`. Then there is no point of setting it to `None` here.

File: Bio/Nexus/Trees.py (hunk 2 above)
Comment (peterjc) [reply Bio/Nexus/Trees.py:598]: We may be OK on CPython 3.6+ (we don't support older Pythons anymore, which had different dict sorting) and recent PyPy.

File: Bio/AlignIO/MafIO.py (hunk 1 above)
Comment (peterjc) [reply Bio/AlignIO/MafIO.py:44]: I'm thinking of the main SeqIO and SearchIO indexing tests, which are skipped when SQLite is missing.

**Comment by mdehoon** [comment]
OK to proceed then?

**Review by peterjc — APPROVED** [review]
Given the automated tests are OK on PyPy, should be fine as is.

**Comment by mdehoon** [comment]
Thanks!
